File: 1834. Single-Threaded CPU/Solution.java
import java.util.PriorityQueue;

class Solution {

    public int[] getOrder(int[][] tasks) {
        int[] order = new int[tasks.length];
        PriorityQueue<Process> priorityQueue = new PriorityQueue<>();
        PriorityQueue<Process> firstPriorityQueue = new PriorityQueue<>();

        for (int i = 0; i < tasks.length; i++) {
            firstPriorityQueue.add(new Process(i, tasks[i][0]));
        }
        int currentInstance = firstPriorityQueue.peek().getTime();
        for (int i = 0; i < order.length;) {
            while (!firstPriorityQueue.isEmpty() && currentInstance >= firstPriorityQueue.peek().getTime()) {
                Process process = firstPriorityQueue.poll();
                priorityQueue.add(new Process(process.getIndex(), tasks[process.getIndex()][1]));
            }
            if (!priorityQueue.isEmpty()) {
                Process process = priorityQueue.poll();
                order[i] = process.getIndex();
                i++;
                currentInstance += process.getTime();
            } else {
                currentInstance = firstPriorityQueue.peek().getTime();
            }
        }
        return order;
    }
}

class Process implements Comparable<Process> {
    private int index;
    private int time;

    public Process(int index, int time) {
        this.index = index;
        this.time = time;
    }

    @Override
    public int compareTo(Process arg0) {
        if (Integer.compare(this.time, arg0.getTime()) == 0)
            return Integer.compare(this.index, arg0.getIndex());
        else
            return Integer.compare(this.time, arg0.getTime());
    }

    public int getIndex() {
        return index;
    }

    public int getTime() {
        return time;
    }

}

File: 1834. Single-Threaded CPU/Solution.py
from queue import PriorityQueue
from typing import List


class Solution:
    def getOrder(self, tasks: List[List[int]]) -> List[int]:
        order = []
        firstPriorityQueue = PriorityQueue()
        secondPriorityQueue = PriorityQueue()
        for i in range(len(tasks)):
            firstPriorityQueue.put(Process(i, tasks[i][0]))

        current = firstPriorityQueue.get()
        secondPriorityQueue.put(
            Process(current.index, tasks[current.index][1]))
        currentTime = current.time

        while (len(order) < len(tasks)):
            while (not firstPriorityQueue.empty()):
                next = firstPriorityQueue.get()
                if (next.time > currentTime):
                    firstPriorityQueue.put(next)
                    break
                else:
                    secondPriorityQueue.put(
                        Process(next.index, tasks[next.index][1]))
                    current = next

            if (not secondPriorityQueue.empty()):
                temp = secondPriorityQueue.get()
                order.append(temp.index)
                currentTime = currentTime + temp.time
            else:
                temp = firstPriorityQueue.get()
                currentTime = temp.time
                firstPriorityQueue.put(temp)

        return order


class Process:
    def __init__(self, index, time) -> None:
        self.index = index
        self.time = time

    def __eq__(self, __o: object) -> bool:
        return self.time == __o.time and self.index == __o.index

    def __lt__(self, __o: object) -> bool:
        if (self.time == __o.time):
            return self.index < __o.index
        else:
            return self.time < __o.time
